Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt

```diff
@@ -23,187 +20,115 @@ import dagger.android.HasAndroidInjector
+        Configuration.Builder()
+            .apply { setMinimumLoggingLevel(android.util.Log.DEBUG) }.build()
+            .let { WorkManager.initialize(this, it) }
 
         NotificationHelper.createNotificationChannel()
+
         // Enable Conscrypt for TLS1.3 Support below API Level 29
         Security.insertProviderAt(Conscrypt.newProvider(), 1)
-        ProcessLifecycleOwner.get().lifecycle.addObserver(this)
-        registerActivityLifecycleCallbacks(this)
 
-        if (BuildConfig.DEBUG) {
-            Timber.plant(Timber.DebugTree())
-        }
-        if ((BuildConfig.FLAVOR == "deviceForTesters" || BuildConfig.DEBUG)) {
-            fileLogger = FileLogger(this)
-        }
+        ProcessLifecycleOwner.get().lifecycle.addObserver(foregroundStateUpdater)
+        registerActivityLifecycleCallbacks(activityLifecycleCallback)
 
-        // notification to test the WakeUpService from Google when the app
-        // was force stopped
+        // notification to test the WakeUpService from Google when the app was force stopped
         BackgroundWorkHelper.sendDebugNotification(
             "Application onCreate", "App was woken up"
         )
-        // Only do this if the background jobs are enabled
-        if (ConnectivityHelper.autoModeEnabled(applicationContext)) {
-            ProcessLifecycleOwner.get().lifecycleScope.launch {
-                // we want a wakelock as the OS does not handle this for us like in the background
-                // job execution
-                val wakeLock = createWakeLock()
-                // we keep a wifi lock to wake up the wifi connection in case the device is dozing
-                val wifiLock = createWifiLock()
-                try {
-                    BackgroundWorkHelper.sendDebugNotification(
-                        "Automatic mode is on", "Check if we have downloaded keys already today"
-                    )
-                    RetrieveDiagnosisKeysTransaction.startWithConstraints()
-                } catch (e: Exception) {
-                    BackgroundWorkHelper.sendDebugNotification(
-                        "RetrieveDiagnosisKeysTransaction failed",
-                        (e.localizedMessage
-                            ?: "Unknown exception occurred in onCreate") + "\n\n" + (e.cause
-                            ?: "Cause is unknown").toString()
-                    )
-                    // retry the key retrieval in case of an error with a scheduled work
-                    BackgroundWorkScheduler.scheduleDiagnosisKeyOneTimeWork()
-                }
-
-                if (wifiLock.isHeld) wifiLock.release()
-                if (wakeLock.isHeld) wakeLock.release()
-            }
+        watchdogService.launch()
+    }
+
+    private val foregroundStateUpdater = object : LifecycleObserver {
+        @OnLifecycleEvent(Lifecycle.Event.ON_START)
+        fun onAppForegrounded() {
+            isAppInForeground = true
+            Timber.v("App is in the foreground")
```

Comment:
Do we actually need these Timber logs? Same below.

Reply:
Was there before I refactored, but I also like it because it gives us more clues about what happened when the testers provide us with a log where there was an issue, we'll know whether the app was in the foreground or background etc. I'm usually a fan of extensive debug logging, huge gains when we encounter heisenbugs :grin:.